lang/en.yml

```
@@ -6,6 +6,7 @@ en:
     REMOVE: Remove
   SilverStripe\Control\ChangePasswordEmail_ss:
     CHANGEPASSWORDFOREMAIL: 'The password for account with email address {email} has been changed. If you didn\''t change your password please change your password using the link below'
+    CHANGEPASSWORDFOREMAIL2: 'The password for account with email address {email} has been changed. If you did not change your password please change your password using the link below'
```

How about using an actual apostrophe instead of a single quote? Then we don't need to introduce a new translation key which I wouldn't do regardless as the meaning is the same.
```suggestion
    CHANGEPASSWORDFOREMAIL: 'The password for account with email address {email} has been changed. If you didn’t change your password please change your password using the link below'
```

---

Changing the original source value will delete all translations in transifex once it's pushed, it means that whenever we want to change something we have to create an entirely new key.

---

> Changing the original source value will delete all translations for the key in transifex once it's pushed

Will it?

I thought we usually don't change the original because the changes we want to make are usually larger than this, and would make the translations inaccurate. Are you sure it deletes them?

---

I'm fairly sure, it would make sense for transifex to do that because if you original was "This is red" and you change it "This is blue" then all existing translations are now wrong, which is worse than them being blank and defaulting to english (which another tool such as automatic browser translations could rectify somewhat)

---

I decided to find out instead of relying on what makes sense: https://help.transifex.com/en/articles/6236849-updating-content#h_ed516d3995

> Any previously submitted translation is removed from the translation area, and the string is considered untranslated.

So Steve's right, we can't change the existing key. The primary reason being that it will remove existing translations.

---

That said, this will be an untranslated string anyway, and we won't be running the translation script against CMS 5.3 anyway.... so the translations will only be removed for CMS 5.4 onwards even if we reuse the same key. So does it really matter?

---

I think we tend to have hard rules like "Create a new key when we change a source translation" to avoid unnecessary discussions each time things like this come up ;-)

---

Right - but I'm suggesting that rule might be outdated. It might be time we revisit and change that rule.

---

If I remember correctly it was because there was some risk someone would run pull translation on to an older branch, e.g. security release. In practice that's probably low risk, however if we start changing source values then we may in the future discuss that actually deleting translations by not creating a new key was in fact a bad idea.

Things are working smoothly now on the translations front though though there were A LOT of issues with translations which spanned years.  It's a complex and somewhat fragile process and I really don't want to alter our process right now.

---

That's fair.

We don't have scheduled meetings for these discussions anymore which is why I wanted to hash this out now - otherwise we could easily say "lets discuss this later" and just never discuss it.

Not changing it because of unknown risk factors seems sensible. I'm okay with leaving the process as-is for now.

---

@michalkleiner Do you still want the apostrophe change made (but using the new key)?

---

The apostrophe is typographically correct-er (sic) but if we need a new key anyway then it doesn't matter much, it was a suggestion purely driven by an attempt to keep the same key.


As a side note FWIW, we use Weblate for Matomo, and there it either can be configured what happens when you update a string (speculating here) or it just keeps it (more likely) and we only introduce a new key if we are changing the meaning substantially. There we decided that an existing but slightly inaccurate translation is better than no translation. Just a view from another project's perspective.

---

If transifex kept the translations that would have been my preference as well. But the documentation implies that the translation is only kept if:
1. The value exactly matches a previously translated value
2. When uploading files _manually_ there's an option to keep translations

There doesn't seem to be an option for keeping translations when uploading source strings through the API.

---

We've had a private issue open for a while to consider an alternative to transifex.
I've now replaced that with a public issue: https://github.com/silverstripe/.github/issues/384
I doubt we'll get to it any time this year, but that's there for discussion.